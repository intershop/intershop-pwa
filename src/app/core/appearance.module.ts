import { NgModule } from '@angular/core';
import { NgbPopoverConfig } from '@ng-bootstrap/ng-bootstrap';
import { ToastrModule } from 'ngx-toastr';

import { IconModule } from './icon.module';
import { ThemeService } from './utils/theme/theme.service';

@NgModule({
  imports: [
    IconModule,
    ToastrModule.forRoot({
      closeButton: true,
      timeOut: 3000,
      positionClass: 'toast-top-full-width', // toast-top-center
      preventDuplicates: true,
    }),
  ],
})
export class AppearanceModule {
  constructor(popoverConfig: NgbPopoverConfig, themeService: ThemeService) {
    popoverConfig.placement = 'top';
    popoverConfig.triggers = 'hover';
    popoverConfig.container = 'body';

    themeService.init();
  }
}
