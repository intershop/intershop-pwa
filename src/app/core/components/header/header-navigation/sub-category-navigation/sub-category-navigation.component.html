<ul *ngIf="category.hasChildren()" class="category-level{{subCategoriesDepth}}" [ngClass]="{'dropdown-menu' : subCategoriesDepth === 1}">
  <li *ngFor="let subcategory of category.children()" class="main-navigation-level{{subCategoriesDepth}}-item" [ngClass]="{'open' : isOpened(subcategory)}">
    <a [routerLink]="['/category', subcategory.uniqueId]" [ngStyle]="{'width' : (!subcategory.hasChildren()) ? '100%' : ''}">{{subcategory.name}}</a>
    <a *ngIf="subcategory.hasChildren()" (click)="toggleOpen(subcategory)" class="dropdown-toggle visible-xs">
      <span class="glyphicon" [ngClass]="isOpened(subcategory) ? 'glyphicon-minus' : 'glyphicon-plus'"></span>
    </a>
    <ish-sub-category-navigation [category]="subcategory" [subCategoriesDepth]="subCategoriesDepth+1"></ish-sub-category-navigation>
  </li>
</ul>
