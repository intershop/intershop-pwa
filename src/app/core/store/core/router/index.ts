// API to access ngrx core router state
export * from './router.selectors';
export * from './router.operators';
