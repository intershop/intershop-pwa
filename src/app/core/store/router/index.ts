export * from './router.actions';
export * from './router.selectors';
export { CoreState } from '../core.state';
