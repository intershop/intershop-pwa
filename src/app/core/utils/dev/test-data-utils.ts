import { CategoryTree, CategoryTreeHelper } from 'ish-core/models/category-tree/category-tree.model';
import { Category } from 'ish-core/models/category/category.model';
import { ContentPageTreeHelper } from 'ish-core/models/content-page-tree/content-page-tree.helper';
import { ContentPageTree, ContentPageTreeElement } from 'ish-core/models/content-page-tree/content-page-tree.model';

export function categoryTree(categories?: Category[]): CategoryTree {
  const tree = CategoryTreeHelper.empty();
  if (categories) {
    return categories.reduce((acc, cat) => CategoryTreeHelper.add(acc, cat), tree);
  }
  return tree;
}

export function pageTree(elements?: ContentPageTreeElement[]): ContentPageTree {
  const tree = ContentPageTreeHelper.empty();
  if (elements) {
    return elements.reduce((acc, cat) => ContentPageTreeHelper.add(acc, cat), tree);
  }
  return tree;
}
