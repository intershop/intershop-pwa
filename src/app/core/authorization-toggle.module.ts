import { ModuleWithProviders, NgModule } from '@angular/core';
import { ReplaySubject } from 'rxjs';
import { map } from 'rxjs/operators';

import { AuthorizationToggleDirective } from './directives/authorization-toggle.directive';
import { AuthorizationToggleService, checkPermission } from './utils/authorization-toggle/authorization-toggle.service';
import { whenTruthy } from './utils/operators';

@NgModule({
  declarations: [AuthorizationToggleDirective],
  exports: [AuthorizationToggleDirective],
})
export class AuthorizationToggleModule {
  private static permissions = new ReplaySubject<string[]>(1);

  static forTesting(...permissions: string[]): ModuleWithProviders<AuthorizationToggleModule> {
    AuthorizationToggleModule.switchTestingPermissions(...permissions);
    return {
      ngModule: AuthorizationToggleModule,
      providers: [
        {
          provide: AuthorizationToggleService,
          useValue: {
            isAuthorizedTo: (permission: string) =>
              AuthorizationToggleModule.permissions.pipe(
                whenTruthy(),
                map(perms => checkPermission(perms, permission))
              ),
          },
        },
      ],
    };
  }

  static switchTestingPermissions(...permissions: string[]) {
    AuthorizationToggleModule.permissions.next(permissions);
  }
}

export { AuthorizationToggleService } from './utils/authorization-toggle/authorization-toggle.service';
export { AuthorizationToggleGuard } from './guards/authorization-toggle.guard';
