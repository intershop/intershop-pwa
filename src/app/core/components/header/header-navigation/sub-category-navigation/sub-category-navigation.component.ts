import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { Category } from '../../../../../models/category/category.model';

@Component({
  selector: 'ish-sub-category-navigation',
  templateUrl: './sub-category-navigation.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})

export class SubCategoryNavigationComponent {

  @Input() category: Category;
  @Input() subCategoriesDepth: number;
}
