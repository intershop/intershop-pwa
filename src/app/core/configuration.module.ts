import { NgModule } from '@angular/core';

import { SPECIAL_HTTP_ERROR_HANDLER } from './interceptors/icm-error-mapper.interceptor';
import { createPaymentErrorHandler } from './utils/http-error/create-payment.error-handler';
import { editPasswordErrorHandler } from './utils/http-error/edit-password.error-handler';
import { LoginUserErrorHandler } from './utils/http-error/login-user.error-handler';
import { requestReminderErrorHandler } from './utils/http-error/request-reminder.error-handler';
import { updatePasswordErrorHandler } from './utils/http-error/update-password.error-handler';

@NgModule({
  providers: [
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: updatePasswordErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useClass: LoginUserErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: requestReminderErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: editPasswordErrorHandler, multi: true },
    { provide: SPECIAL_HTTP_ERROR_HANDLER, useValue: createPaymentErrorHandler, multi: true },
  ],
})
export class ConfigurationModule {}
