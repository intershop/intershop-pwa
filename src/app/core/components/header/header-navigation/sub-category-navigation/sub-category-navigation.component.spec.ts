import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { createCategoryView } from '../../../../../models/category-view/category-view.model';
import { Category } from '../../../../../models/category/category.model';
import { categoryTree } from '../../../../../utils/dev/test-data-utils';
import { SubCategoryNavigationComponent } from './sub-category-navigation.component';

describe('SubCategory Navigation Component', () => {
  let fixture: ComponentFixture<SubCategoryNavigationComponent>;
  let component: SubCategoryNavigationComponent;
  let element: HTMLElement;

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        imports: [RouterTestingModule],
        declarations: [SubCategoryNavigationComponent],
      }).compileComponents();
    })
  );

  beforeEach(() => {
    fixture = TestBed.createComponent(SubCategoryNavigationComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;

    const tree = categoryTree([
      { uniqueId: 'A', name: 'CAT_A', categoryPath: ['A'] },
      { uniqueId: 'A.1', name: 'CAT_A1', categoryPath: ['A', 'A.1'] },
      { uniqueId: 'A.2', name: 'CAT_A2', categoryPath: ['A', 'A.2'] },
      { uniqueId: 'A.1.a', name: 'CAT_A1a', categoryPath: ['A', 'A.1', 'A.1.a'] },
    ] as Category[]);

    component.category = createCategoryView(tree, 'A');
    component.subCategoriesDepth = 0;
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(() => fixture.detectChanges()).not.toThrow();
    expect(element).toMatchSnapshot();
  });
});
