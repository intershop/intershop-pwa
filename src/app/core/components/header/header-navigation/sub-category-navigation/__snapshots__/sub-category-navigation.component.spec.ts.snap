// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sub Category Navigation Component should be created 1`] = `
<div
  id="root0"
  ng-version="6.0.7"
>
  
  <ul
    class="category-level1 dropdown-menu"
    ng-reflect-klass="category-level1"
    ng-reflect-ng-class="[object Object]"
  >
    
    <li
      class="main-navigation-level1-item"
      ng-reflect-klass="main-navigation-level1-item"
      ng-reflect-ng-class="[object Object]"
    >
      <a
        href="/category/A.1"
        ng-reflect-ng-style="[object Object]"
        ng-reflect-router-link="/category,A.1"
      >
        CAT_A1
      </a>
      
      <a
        class="dropdown-toggle visible-xs"
      >
        <span
          class="glyphicon glyphicon-plus"
          ng-reflect-klass="glyphicon"
          ng-reflect-ng-class="glyphicon-plus"
        />
      </a>
      
      <ish-sub-category-navigation
        ng-reflect-category="[object Object]"
        ng-reflect-sub-categories-depth="2"
      >
        
        <ul
          class="category-level2"
          ng-reflect-klass="category-level2"
          ng-reflect-ng-class="[object Object]"
        >
          
          <li
            class="main-navigation-level2-item"
            ng-reflect-klass="main-navigation-level2-item"
            ng-reflect-ng-class="[object Object]"
          >
            <a
              href="/category/A.1.a"
              ng-reflect-ng-style="[object Object]"
              ng-reflect-router-link="/category,A.1.a"
            >
              CAT_A1a
            </a>
            
            <a
              class="dropdown-toggle visible-xs"
            >
              <span
                class="glyphicon glyphicon-plus"
                ng-reflect-klass="glyphicon"
                ng-reflect-ng-class="glyphicon-plus"
              />
            </a>
            
          </li>
        </ul>
      </ish-sub-category-navigation>
    </li>
    <li
      class="main-navigation-level1-item"
      ng-reflect-klass="main-navigation-level1-item"
      ng-reflect-ng-class="[object Object]"
    >
      <a
        href="/category/A.2"
        ng-reflect-ng-style="[object Object]"
        ng-reflect-router-link="/category,A.2"
        style="width: 100%;"
      >
        CAT_A2
      </a>
      
      
      <ish-sub-category-navigation
        ng-reflect-category="[object Object]"
        ng-reflect-sub-categories-depth="2"
      >
        
      </ish-sub-category-navigation>
    </li>
  </ul>
</div>
`;
